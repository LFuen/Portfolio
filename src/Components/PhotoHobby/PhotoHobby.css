


.flexRow {
    display: flex;
    flex-direction: row;
    margin-bottom: 5px;
    align-items: center;
    justify-content: center;
}

img {
    width: 100%;
}

.margin {
    margin-top: 40px;
    margin-bottom: 100px;
}

div.bg-color {
    padding: 25px 10px 25px 10px;
}

.item {
    flex: 1;
    margin: 10px;
}


@keyframes bob {
    0% {
        transform: translateY(-8px);
    }
    50% {
        transform: translateY(-4px);
    }
    100% {
        transform: translateY(-8px);
    }
}

@keyframes float {
    100% {
        transform: translateY(-8px);
    }
}

img:hover {
    animation-name: float, bob;
    animation-duration: .3s, 1.5s;
    animation-delay: 0s, .3s;
    animation-timing-function: ease-out, ease-in-out;
    animation-iteration-count: 1, infinite;
    animation-fill-mode: forwards;
    animation-direction: normal, alternate;
}

@media screen and (max-width: 1020px) {
    .flexRow{
    margin: auto;
    flex-direction: column;
    font-size: medium;
    }
    
    h1{
        font-size: 40px !important;
    }

    .margin{
        margin-top: 20px;
    }
}