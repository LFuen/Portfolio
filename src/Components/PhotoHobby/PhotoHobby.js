import {Component} from 'react'
import './PhotoHobby.css'


class PhotoHobby extends Component {

    render() {

        return (
            <div className='bg-color margin mobile'>
                <div className='flexRow'>
                    <div className='item'>
                        <a 
                            className='picture' 
                            href='https://lfuen.github.io/portimages/greatFalls.jpeg' 
                            target='_blank'
                            rel='noreferrer'
                            title='Great Falls'>
                            
                            <img 
                                src='https://lfuen.github.io/portimages/greatFalls.jpeg' // {photos[active]}
                                alt='Great Falls' 
                                width='400'/>
                        </a>
                    </div>
                    <div className='item'>
                        <a 
                            className='picture' 
                            href='https://lfuen.github.io/portimages/reflection.jpeg' 
                            target='_blank'
                            rel='noreferrer'
                            title='Reflection'>
                            
                            <img 
                                src='https://lfuen.github.io/portimages/reflection.jpeg' 
                                alt='Reflection' 
                                width='400'/>
                        </a>
                    </div>
                    <div className='item'>
                        <a 
                            className='picture' 
                            href='https://lfuen.github.io/portimages/road.jpeg' 
                            target='_blank'
                            rel='noreferrer'
                            title='Sherrill Drive'>
                            
                            <img 
                                src='https://lfuen.github.io/portimages/road.jpeg' 
                                alt='Sherrill Drive' 
                                width='400'/>
                        </a>
                    </div>
                    <div className='item'>
                        <a 
                            className='picture' 
                            href='https://lfuen.github.io/portimages/waterfall.jpeg' 
                            target='_blank'
                            rel='noreferrer'
                            title='Cunningham Falls'>
                            
                            <img 
                                src='https://lfuen.github.io/portimages/waterfall.jpeg' 
                                alt='Cunningham Falls' 
                                width='400'/>
                        </a>
                    </div>
                    <div className='item'>    
                        <a 
                            className='picture' 
                            href='https://lfuen.github.io/portimages/stream.jpeg' 
                            target='_blank'
                            rel='noreferrer'
                            title='Pinehurst Branch'>
                            
                            <img 
                                src='https://lfuen.github.io/portimages/stream.jpeg' 
                                alt='Pinehurst Branch' 
                                width='400'/>
                        </a>
                    </div>

                </div>
            </div>
        )
    }
}

export default PhotoHobby

