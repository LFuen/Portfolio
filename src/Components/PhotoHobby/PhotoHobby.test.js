import React from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import PhotoHobby from './PhotoHobby'


describe(`PhotoHobby component`, () => {
    describe(`Snapshot`, () => {
        it(`renders the UI as expected`, () => {
            const photo = renderer.create(
                <BrowserRouter>
                    <PhotoHobby />
                </BrowserRouter>).toJSON()
                expect(photo).toMatchSnapshot()
        })
    })

    describe(`Smoke test`, () => {
        it(`renders without crashing`, () => {
            const div = document.createElement('div')

            ReactDOM.render(
                <BrowserRouter>
                    <PhotoHobby />
                </BrowserRouter>,
                div
            )
            ReactDOM.unmountComponentAtNode(div)
        })
    })
})