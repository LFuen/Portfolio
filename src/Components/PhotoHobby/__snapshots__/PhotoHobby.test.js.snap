// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoHobby component Snapshot renders the UI as expected 1`] = `
<div
  className="bg-color margin"
>
  <div
    className="flexRow"
  >
    <div
      className="item"
    >
      <a
        className="picture"
        href="https://lfuen.github.io/portimages/greatFalls.jpeg"
        rel="noreferrer"
        target="_blank"
        title="Great Falls"
      >
        <img
          alt="Great Falls"
          src="https://lfuen.github.io/portimages/greatFalls.jpeg"
          width="400"
        />
      </a>
    </div>
    <div
      className="item"
    >
      <a
        className="picture"
        href="https://lfuen.github.io/portimages/reflection.jpeg"
        rel="noreferrer"
        target="_blank"
        title="Reflection"
      >
        <img
          alt="Reflection"
          src="https://lfuen.github.io/portimages/reflection.jpeg"
          width="400"
        />
      </a>
    </div>
    <div
      className="item"
    >
      <a
        className="picture"
        href="https://lfuen.github.io/portimages/road.jpeg"
        rel="noreferrer"
        target="_blank"
        title="Sherrill Drive"
      >
        <img
          alt="Sherrill Drive"
          src="https://lfuen.github.io/portimages/road.jpeg"
          width="400"
        />
      </a>
    </div>
    <div
      className="item"
    >
      <a
        className="picture"
        href="https://lfuen.github.io/portimages/waterfall.jpeg"
        rel="noreferrer"
        target="_blank"
        title="Cunningham Falls"
      >
        <img
          alt="Cunningham Falls"
          src="https://lfuen.github.io/portimages/waterfall.jpeg"
          width="400"
        />
      </a>
    </div>
    <div
      className="item"
    >
      <a
        className="picture"
        href="https://lfuen.github.io/portimages/stream.jpeg"
        rel="noreferrer"
        target="_blank"
        title="Pinehurst Branch"
      >
        <img
          alt="Pinehurst Branch"
          src="https://lfuen.github.io/portimages/stream.jpeg"
          width="400"
        />
      </a>
    </div>
  </div>
</div>
`;
